### DISCUSSION TITLE
Callback on inline button not working

### DISCUSSION THREAD

**Discussion by Randommist** [aiogram/aiogram#1239] [General]
A very strange situation. For some reason, the callback handler on the inline button stopped working for me and I can’t even understand what the problem is. I already tried different versions of aiogram even tried version 3.0, tried different versions of python and ran the code on different machines but nothing helped. I even tried to use a third-party library to work with the telegram api, but that didn't help either, so I'm not even sure if the problem is in aiogram. Example code below

```
from aiogram import types, executor, Dispatcher, Bot
from aiogram.types import InlineKeyboardButton, InlineKeyboardMarkup


bot = Bot("TOKEN")
dp = Dispatcher(bot)


@dp.message_handler(commands=["start"])
async def cmd_start(message: types.Message) -> None:
    kb = InlineKeyboardMarkup()
    kb.insert(InlineKeyboardButton(text="test", callback_data="test"))

    await bot.send_message(message.from_user.id, text="some text", reply_markup=kb)


@dp.callback_query_handler(text="test")
async def test(callback: types.CallbackQuery) -> None:
    await callback.answer("pressed")


if __name__ == "__main__":
    executor.start_polling(dp)
```

**Comment by Randommist** [comment]
I decided to test this code again and this is what I noticed.  When I change the token to another bot, it starts processing the callback.  But on the previous bot, the callback handler still does not work, although regular messages from the bot come. The solution was to create a new bot.  Although I don't know why this is happening

**Comment by Olegt0rr** [comment]
## Telegram Bot API behavior

Telegram Bot API method [getUpdates](https://core.telegram.org/bots/api#getupdates) has a param `allowed_updates`. If param is not specified, **the previous setting will be used**... 

## aiogram 3.x behavior

Polling in aiogram v3.x by default uses auto-discovery (#1182) of update types. 
If you run an app with message-only handlers, polling system will provide `allowed_updates` field with "message" only by default.
Then Telegram Bot API won't send you callback_query or any other update type...

## aiogram 2.x behavior

Polling in aiogram v2.x is stupidly simple and predictable: 
if you don't set `allowed_updates` – it will not include it in polling requests.

## How to surprise yourself

1. Register new bot.
2. Run your bot with aiogram v3 ([echo example](https://github.com/aiogram/aiogram/blob/dev-3.x/examples/echo_bot.py)). It works! Stop it.
3. Run you bot with aiogram v2 ([code](https://github.com/aiogram/aiogram/discussions/1239#discussion-5462556) provided by discussion author, with inline-buttons). Inline-buttons don't work!

The reason for this behavior is that the Telegram Bot API cached your `allowed_updates = [ "message" ]` setting from step 2 (thanks for auto-discover) and applied it in step 3.

## Workaround

Always use `allowed_updates` field with expected list of event types

**Reply by Randommist** [reply]
Thank you for such a detailed explanation. Now everything is in place 🎉
